File: 0684/sesión12 - WA/SoftVen/SoftVenPersistance/DAOImpl/ProductoDAOImpl.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data.Common;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftVen.DAO;
using SoftVen.DAOImpl;
using SoftVen.DAOImpl.Util;
using SoftVen.Model;

namespace SoftVenPersistance.DAOImpl
{
    public class ProductoDAOImpl: DAOImplBase, ProductoDAO
    {
        private ProductosDTO producto;

        public ProductoDAOImpl(): base("VEN_PRODUCTOS")
        {            
            this.retornarLlavePrimaria = true;
            this.producto = null;
        }

        protected override void ConfigurarListaDeColumnas()
        {
            this.listaColumnas.Add(new Columna("PRODUCTO_ID", true, true));
            this.listaColumnas.Add(new Columna("NOMBRE",  false, false));
            this.listaColumnas.Add(new Columna("DESCRIPCION", false, false));
            this.listaColumnas.Add(new Columna("PRECIO", false, false));
        }

        protected override void IncluirValorDeParametrosParaInsercion()
        {            
            this.AgregarParametro("@NOMBRE", this.producto.Nombre);
            this.AgregarParametro("@DESCRIPCION", this.producto.Descripcion);
            this.AgregarParametro("@PRECIO", this.producto.Precio);
        }
        
        protected override void IncluirValorDeParametrosParaModificacion()
        {
            //this.statement.setString(1, this.almacen.getNombre());
            //this.statement.setInt(2, this.almacen.getAlmacen_central() ? 1 : 0);
            //this.statement.setInt(3, this.almacen.getAlmacenId());
        }

        protected override void IncluirValorDeParametrosParaEliminacion()
        {
            //this.statement.setInt(1, this.almacen.getAlmacenId());
        }

        protected override void IncluirValorDeParametrosParaObtenerPorId()
        {
            //this.statement.setInt(1, this.almacen.getAlmacenId());
        }

        protected override void InstanciarObjetoDelResultSet(DbDataReader lector)
        {
            this.producto = new ProductosDTO();
            this.producto.ProductoId = this.lector.GetInt32(0);
            this.producto.Nombre = this.lector.GetString(1);
            this.producto.Descripcion = this.lector.GetString(2);
            this.producto.Precio = this.lector.GetDouble(3);
        }

        protected override void LimpiarObjetoDelResultSet()
        {
            this.producto = null;
        }

        protected override void AgregarObjetoALaLista(BindingList<Object> lista, DbDataReader lector)
        {
            this.InstanciarObjetoDelResultSet(lector);
            lista.Add(this.producto);
        }

        public int Insertar(ProductosDTO producto)
        {
            this.producto = producto;
            return base.Insertar();
        }

        public int Modificar(ProductosDTO producto)
        {
            this.producto = producto;
            return base.Modificar();
        }

        public int Eliminar(ProductosDTO producto)
        {
            this.producto = producto;
            return base.Eliminar();
        }

        public ProductosDTO ObtenerPorId(int productoId)
        {
            this.producto = new ProductosDTO();
            this.producto.ProductoId = productoId;
            base.ObtenerPorId();
            return this.producto;
        }

        public new BindingList<ProductosDTO> ListarTodos()
        {
            BindingList<Object> lista;
            lista = base.ListarTodos();
            BindingList<ProductosDTO> retorno = new BindingList<ProductosDTO>();
            foreach (ProductosDTO objecto in lista)
            {
                retorno.Add(objecto);
            }
            return retorno;
        }
    }
}

File: 0684/sesión19 - WS - SOAP/Back-End/SoftVenPersistance/src/main/java/pe/edu/pucp/softven/daoImp/ProductoDAOImpl.java
package pe.edu.pucp.softven.daoImp;

import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import pe.edu.pucp.softven.dao.ProductoDAO;
import pe.edu.pucp.softven.daoImp.util.Columna;
import pe.edu.pucp.softven.daoImp.util.Tipo_Dato;
import pe.edu.pucp.softven.model.ProductosDTO;

public class ProductoDAOImpl extends DAOImplBase implements ProductoDAO {

    private ProductosDTO producto;

    public ProductoDAOImpl() {
        super("VEN_PRODUCTOS");
        this.retornarLlavePrimaria = true;
        this.producto = null;
    }

    @Override
    protected void configurarListaDeColumnas() {
        this.listaColumnas.add(new Columna("PRODUCTO_ID", Tipo_Dato.ENTERO, true, true));
        this.listaColumnas.add(new Columna("NOMBRE", Tipo_Dato.CADENA, false, false));
        this.listaColumnas.add(new Columna("DESCRIPCION", Tipo_Dato.CADENA, false, false));
        this.listaColumnas.add(new Columna("PRECIO", Tipo_Dato.REAL, false, false));
    }

    @Override
    protected void incluirValorDeParametrosParaInsercion() throws SQLException {
        this.statement.setString(1, this.producto.getNombre());
        this.statement.setString(2, this.producto.getDescripcion());
        this.statement.setDouble(3, this.producto.getPrecio());
    }

    @Override
    protected void incluirValorDeParametrosParaModificacion() throws SQLException {
        this.statement.setString(1, this.producto.getNombre());
        this.statement.setString(2, this.producto.getDescripcion());
        this.statement.setDouble(3, this.producto.getPrecio());
        this.statement.setInt(4, this.producto.getProductoId());
    }

    @Override
    protected void incluirValorDeParametrosParaEliminacion() throws SQLException {
        this.statement.setInt(1, this.producto.getProductoId());
    }

    @Override
    protected void incluirValorDeParametrosParaObtenerPorId() throws SQLException {
        this.statement.setInt(1, this.producto.getProductoId());
    }

    @Override
    protected void instanciarObjetoDelResultSet() throws SQLException {
        this.producto = new ProductosDTO();
        this.producto.setProductoId(this.resultSet.getInt("PRODUCTO_ID"));
        this.producto.setNombre(this.resultSet.getString("NOMBRE"));
        this.producto.setDescripcion(this.resultSet.getString("DESCRIPCION"));
        this.producto.setPrecio(this.resultSet.getDouble("PRECIO"));
    }

    @Override
    protected void limpiarObjetoDelResultSet() {
        this.producto = null;
    }

    @Override
    protected void agregarObjetoALaLista(List lista) throws SQLException {
        this.instanciarObjetoDelResultSet();
        lista.add(this.producto);
    }

    @Override
    public Integer insertar(ProductosDTO producto) {
        this.producto = producto;
        return super.insertar();
//        int resultado = 0;
//        try {
//            this.conexion = DBManager.getInstance().getConnection();
//            this.conexion.setAutoCommit(false);
//            String sql = "INSERT INTO VEN_PRODUCTOS (NOMBRE, DESCRIPCION, PRECIO) VALUES (?,?,?)";
//            this.statement = this.conexion.prepareCall(sql);
//            this.statement.setString(1, producto.getNombre());
//            this.statement.setString(2, producto.getDescripcion());
//            this.statement.setDouble(3, producto.getPrecio());
//            //resultado = this.statement.executeUpdate();
//            this.statement.executeUpdate();
//            resultado = this.retornarUltimoAutoGenerado();
//            this.conexion.commit();
//        } catch (SQLException ex) {
//            System.err.println("Error al intentar insertar - " + ex);
//            try {
//                if (this.conexion != null) {
//                    this.conexion.rollback();
//                }
//            } catch (SQLException ex1) {
//                System.err.println("Error al hacer rollback - " + ex1);
//            }
//        } finally {
//            try {
//                if (this.conexion != null) {
//                    this.conexion.close();
//                }
//            } catch (SQLException ex) {
//                System.err.println("Error al cerrar la conexión - " + ex);
//            }
//        }
//        return resultado;
    }

    @Override
    public ProductosDTO obtenerPorId(Integer productoId) {
        this.producto = new ProductosDTO();
        this.producto.setProductoId(productoId);
        super.obtenerPorId();
        return this.producto;
//        ProductosDTO producto = null;
//        try {
//            this.conexion = DBManager.getInstance().getConnection();
//            String sql = "SELECT PRODUCTO_ID, NOMBRE, DESCRIPCION, PRECIO FROM VEN_PRODUCTOS WHERE PRODUCTO_ID = ?";
//            this.statement = this.conexion.prepareCall(sql);
//            this.statement.setInt(1, productoId);
//            this.resultSet = this.statement.executeQuery();
//            if (this.resultSet.next()) {
//                producto = new ProductosDTO();
//                producto.setProductoId(this.resultSet.getInt("PRODUCTO_ID"));
//                producto.setNombre(this.resultSet.getString("NOMBRE"));
//                producto.setDescripcion(this.resultSet.getString("DESCRIPCION"));
//                producto.setPrecio(this.resultSet.getDouble("PRECIO"));
//            }
//        } catch (SQLException ex) {
//            System.err.println("Error al intentar obtenerPorId - " + ex);
//        } finally {
//            try {
//                if (this.conexion != null) {
//                    this.conexion.close();
//                }
//            } catch (SQLException ex) {
//                System.err.println("Error al cerrar la conexión - " + ex);
//            }
//        }
//        return producto;
    }

    @Override
    public ArrayList<ProductosDTO> listarTodos() {
        return (ArrayList<ProductosDTO>) super.listarTodos();
//        ArrayList<ProductosDTO> listaProductos = new ArrayList<>();
//        try {
//            this.conexion = DBManager.getInstance().getConnection();
//            String sql = "SELECT PRODUCTO_ID, NOMBRE, DESCRIPCION, PRECIO FROM VEN_PRODUCTOS";
//            this.statement = this.conexion.prepareCall(sql);
//            this.resultSet = this.statement.executeQuery();
//            while (this.resultSet.next()) {
//                ProductosDTO producto = new ProductosDTO();
//                producto.setProductoId(this.resultSet.getInt("PRODUCTO_ID"));
//                producto.setNombre(this.resultSet.getString("NOMBRE"));
//                producto.setDescripcion(this.resultSet.getString("DESCRIPCION"));
//                producto.setPrecio(this.resultSet.getDouble("PRECIO"));
//                listaProductos.add(producto);
//            }
//        } catch (SQLException ex) {
//            System.err.println("Error al intentar listarTodos - " + ex);
//        } finally {
//            try {
//                if (this.conexion != null) {
//                    this.conexion.close();
//                }
//            } catch (SQLException ex) {
//                System.err.println("Error al cerrar la conexión - " + ex);
//            }
//        }
//        return listaProductos;
    }

    @Override
    public Integer modificar(ProductosDTO producto) {
        this.producto = producto;
        return super.modificar();
//        int resultado = 0;
//        try {
//            this.conexion = DBManager.getInstance().getConnection();
//            this.conexion.setAutoCommit(false);
//            String sql = "UPDATE VEN_PRODUCTOS SET NOMBRE=?, DESCRIPCION=?, PRECIO=? WHERE PRODUCTO_ID=?";
//            this.statement = this.conexion.prepareCall(sql);
//            this.statement.setString(1, producto.getNombre());
//            this.statement.setString(2, producto.getDescripcion());
//            this.statement.setDouble(3, producto.getPrecio());
//            this.statement.setInt(4, producto.getProductoId());
//
//            resultado = this.statement.executeUpdate();
//            this.conexion.commit();
//        } catch (SQLException ex) {
//            System.err.println("Error al intentar modificar - " + ex);
//            try {
//                if (this.conexion != null) {
//                    this.conexion.rollback();
//                }
//            } catch (SQLException ex1) {
//                System.err.println("Error al hacer rollback - " + ex1);
//            }
//        } finally {
//            try {
//                if (this.conexion != null) {
//                    this.conexion.close();
//                }
//            } catch (SQLException ex) {
//                System.err.println("Error al cerrar la conexión - " + ex);
//            }
//        }
//        return resultado;
    }

    @Override
    public Integer eliminar(ProductosDTO producto) {
        this.producto = producto;
        return super.eliminar();
//        int resultado = 0;
//        try {
//            this.conexion = DBManager.getInstance().getConnection();
//            this.conexion.setAutoCommit(false);
//            String sql = "DELETE FROM VEN_PRODUCTOS WHERE PRODUCTO_ID=?";
//            this.statement = this.conexion.prepareCall(sql);
//            this.statement.setInt(1, producto.getProductoId());
//            resultado = this.statement.executeUpdate();
//            this.conexion.commit();
//        } catch (SQLException ex) {
//            System.err.println("Error al intentar eliminar - " + ex);
//            try {
//                if (this.conexion != null) {
//                    this.conexion.rollback();
//                }
//            } catch (SQLException ex1) {
//                System.err.println("Error al hacer rollback - " + ex1);
//            }
//        } finally {
//            try {
//                if (this.conexion != null) {
//                    this.conexion.close();
//                }
//            } catch (SQLException ex) {
//                System.err.println("Error al cerrar la conexión - " + ex);
//            }
//        }
//        return resultado;
    }

}
